test/unit_test/simulator.h

```diff
@@ -5,6 +5,10 @@
 
 namespace simulator
 {
+// maps an int: # learned examples
+// with a function to 'test' at that point in time in the simulator
+using callback_map = typename std::map<int, std::function<bool(vw*)>>;
```

Comment:
it will be later i think, or what makes you say that?

Reply:
`std::map` is not ambiguous so you don't need to tell the compiler is a type? I'm not 100% so leaving it here is fine

Reply:
yea the benefit is low, i was thinking for default arg to some function. It looks a bit cleaner with a short typedef

Reply:
`typename` shouldn't be needed since there isn't a need to specify that this is a type, I think it is mostly used with templates to distinguish any ambiguity